spring-grpc-spring-boot-autoconfigure/src/test/java/org/springframework/grpc/autoconfigure/server/GrpcServletAutoConfigurationTests.java

```diff
@@ -90,7 +74,40 @@ void whenNoBindableServicesRegisteredAutoConfigurationIsSkipped() {
 
 	@Test
 	void whenWebApplicationServletIsAutoConfigured() {
-		this.contextRunner().run((context) -> assertThat(context).getBean(ServletRegistrationBean.class).isNotNull());
+		this.contextRunner().run((context) -> {
+			assertThat(context).getBean(ServletRegistrationBean.class)
+				.isNotNull()
+				.extracting("servlet")
+				.isInstanceOf(GrpcServlet.class)
+				.extracting("servletAdapter")
+				.isInstanceOf(ServletAdapter.class)
+				.extracting("maxInboundMessageSize")
+				.isEqualTo(GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE);
+		});
+	}
+
+	@Test
+	void whenCustomizerIsRegistered() {
+		AtomicBoolean invoked = new AtomicBoolean(false);
+		ServerBuilderCustomizer<ServletServerBuilder> customizer = serverBuilder -> invoked.set(true);
+		this.contextRunner().withBean(ServerBuilderCustomizer.class, () -> customizer).run(context -> {
+			assertThat(context).getBean(ServletRegistrationBean.class).isNotNull();
+			assertThat(invoked.get()).isTrue();
+		});
+	}
+
+	@Test
+	void whenMaxInboundMessageSizeIsConfigured() {
```

Comment:
As mentioned above, I think adding another variant here that tests the property value when set and when not set would be good. Something like:

```java
	@Test
	void whenMaxInboundMessageSizeIsSetThenItIsUsed() {
		this.contextRunner()
			.withPropertyValues("spring.grpc.server.max-inbound-message-size=10KB")
			.run(context -> assertThat(context).getBean(ServletRegistrationBean.class)
				.hasFieldOrPropertyWithValue("servlet.servletAdapter.maxInboundMessageSize",
						Math.toIntExact(DataSize.ofKilobytes(10).toBytes())));
	}

	@Test
	void whenMaxInboundMessageSizeIsNotSetThenDefaultIsUsed() {
		this.contextRunner()
			.run((context) -> assertThat(context).getBean(ServletRegistrationBean.class)
				.hasFieldOrPropertyWithValue("servlet.servletAdapter.maxInboundMessageSize",
						GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE));
	}
```
Also, notice the use of `hasFieldOrPropertyWithValue` to avoid the nested drill-down to get to the property value. 

Wdyt?

Reply:
make sense. fixed in 2f50e5c27732722dd6027127135d1673f1503a59